pkg/client/add_events.go

```diff
@@ -279,22 +273,26 @@ func (client *DataSetClient) Shutdown() error {
 	client.Logger.Info("Shutting down - publishing all buffers")
 	client.publishAllBuffers()
 
-	// do wait for all buffers to be processed
+	// reinitialize expBackoff with MaxElapsedTime based on actually elapsed time of processing (previous phase)
+	processingElapsed := time.Since(processingStart)
+	remainingShutdownTimeout := shutdownTimeout - processingElapsed
+	expBackoff = backoff.ExponentialBackOff{
+		InitialInterval:     client.Config.BufferSettings.RetryInitialInterval,
+		RandomizationFactor: client.Config.BufferSettings.RetryRandomizationFactor,
+		Multiplier:          client.Config.BufferSettings.RetryMultiplier,
+		MaxInterval:         client.Config.BufferSettings.RetryMaxInterval,
```

Comment:
I believe, that here we should use the same trick with the min
```suggestion
		MaxInterval:         minDuration(client.Config.BufferSettings.RetryMaxElapsedTime, remainingShutdownTimeout)
```

Reply:
I ma not sure why. Event if `RetryMaxElapsedTime` or `RetryShutdownTimeout` is < `MaxInterval` the interval is exponentially increased so we never get to size of defined interval and stop earlier no matter of `MaxInterval` config value.

Reply:
It makes sense, that the backoff implementation is smart, so if the retries would be: 1, 2, 4, 8, 16 - and the max elapsed time would be 20, then it would stop after 5s and not wait all the 16s before next check.